### PR TITLE
Build Docker Containers for Linux Wheels

### LINKED ISSUE #6294: Kubernetes Docker Container untagged
The container used by the kubernetes manifests (rayproject/autoscaler) doesn't have any useful tags. Furthermore the contents of the container are actually built using github.com/edoakes/ray and the container is two months old.

It might be better to properly tag the containers with versions and ensure they are built from the main repo (by travis) so that there aren't small differences in the code and it stays up to date.

### PR DESCRIPTION
**Pull request by simon-mo**
Note the limitation, we currently:
- Only build python 3.6 wheels
- CPU
- With a predefined set of requirements (mostly drawing from Tune's requirement)

Closes #6294

### CODE REVIEW

**Comment by simon-mo** [comment]
Example log here https://travis-ci.com/ray-project/ray/jobs/259983442#L241-L747

File: docker/autoscaler/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+gym[atari]
```
Comment (edoakes) [inline docker/autoscaler/requirements.txt]: +kubernetess, boto3

File: docker/autoscaler/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM ray-project/base-deps
+ARG WHEEL_PATH
+ARG WHEEL_NAME
+
+# We have to uninstall wrapt this way for Tensorflow compatibility
+RUN conda remove -y wrapt
+COPY requirements.txt .
+RUN pip install -r requirements.txt
+
```
Comment (edoakes) [inline docker/autoscaler/Dockerfile]: let's also install curl, tmux, screen, rsync

**Review by edoakes — CHANGES_REQUESTED** [review]
Can you make sure to run some tests using the autoscaler on k8s with an image you build using the script to make sure we have everything needed?

**Review by edoakes — APPROVED** [review]
LGTM please test as described above before merging

**Comment by simon-mo** [comment]
Tested on autoscaler. With a minor fix (environment variable for click), it worked like a charm. Next step is to reduce the image size and support GPU, other version of python as well.

**Pull request merged**